src/test/java/com/exasol/adapter/dialects/db2/DB2SqlDialectIT.java

```diff
@@ -0,0 +1,233 @@
+        final Bucket bucket = EXASOL.getDefaultBucket();
+        final Path pathToSettingsFile = Path.of("src", "test", "resources", JDBC_DRIVER_CONFIGURATION_FILE_NAME);
+        bucket.uploadFile(JDBC_DRIVER_PATH, "drivers/jdbc/" + JDBC_DRIVER_NAME);
+        bucket.uploadFile(pathToSettingsFile, "drivers/jdbc/" + JDBC_DRIVER_CONFIGURATION_FILE_NAME);
+    }
+
+    @AfterAll
+    static void afterAll() throws SQLException {
+        dropAll(adapterScript, adapterSchema);
+        adapterScript = null;
+        adapterSchema = null;
+        exasolConnection.close();
+        db2Connection.close();
+    }
+
+    private static void dropAll(final DatabaseObject... databaseObjects) {
+        for (final DatabaseObject databaseObject : databaseObjects) {
+            if (databaseObject != null) {
+                databaseObject.drop();
+            }
+        }
+    }
+
+    @Test
+    void testSmallintMapping() throws SQLException {
+        final String table = createSingleColumnTable("SMALLINT", List.of("-32768", "32767"));
+        assertVirtualTableContents(table, table().row(-32768).row(32767).matches());
+    }
+
+    @Test
+    void testIntegerMapping() throws SQLException {
+        final String table = createSingleColumnTable("INT", List.of("-2147483648", "2147483647"));
+        assertVirtualTableContents(table, table().row(-2147483648).row(2147483647).matches(NO_JAVA_TYPE_CHECK));
+    }
+
+    @Test
+    void testBigintMapping() throws SQLException {
+        final String table = createSingleColumnTable("BIGINT", List.of("-9223372036854775808", "9223372036854775807"));
+        assertVirtualTableContents(table,
+                table().row(-9223372036854775808L).row(9223372036854775807L).matches(NO_JAVA_TYPE_CHECK));
+    }
+
+    @Test
+    void testDecimalMapping() throws SQLException {
+        final String table = createSingleColumnTable("DECIMAL(31, 5)", List.of("-123.456", "123.456"), "DECIMAL");
+        assertVirtualTableContents(table, table().row(-123.456).row(123.456).matches(NO_JAVA_TYPE_CHECK));
+    }
+
+    @Test
+    void testRealMapping() throws SQLException {
+        final String table = createSingleColumnTable("REAL", List.of("-123.457", "0.0001245"));
+        assertVirtualTableContents(table,
+                table().row(-123.45700073242188).row(1.2450000212993473E-4).matches(NO_JAVA_TYPE_CHECK));
+    }
+
+    @Test
+    void testDoubleMapping() throws SQLException {
+        final String table = createSingleColumnTable("DOUBLE", List.of("-1000", "14.568001"));
+        assertVirtualTableContents(table, table().row(-1000).row(14.568001).matches(NO_JAVA_TYPE_CHECK));
+    }
+
+    @Test
+    void testCharMapping() throws SQLException {
+        final String table = createSingleColumnTable("CHAR(15)", List.of("'This is a test'", "'Hello'"), "CHAR");
+        assertVirtualTableContents(table, table().row("This is a test ").row("Hello          ").matches());
```

Comment:
why do you test everything with two rows?

Reply:
Why not? :)